docs/architecture/adr-002-api-cleanup.md

```diff
@@ -0,0 +1,96 @@
+# ADR ADR-002: API Cleanup to Support Store V2
+
+## Changelog
+
+* 2023-06-06: First draft
+
+## Status
+
+DRAFT
+
+## Abstract
+
+This ADR proposes a cleanup of the API to support [Store V2](https://github.com/cosmos/cosmos-sdk/blob/main/docs/architecture/adr-065-store-v2.md).
+
+There are some proposals for the speedup of the `Commit` by the async writes. See the [Discussion](https://github.com/cosmos/cosmos-sdk/issues/16173) for more details.
+
+There is a lot of legacy code in the SDK that is not used anymore and can be removed. See the [Discussion](https://github.com/cosmos/iavl/issues/737) for more details. 
+
+## Context
+
+The introduction of `Store V2` separates SS (State Storage) and SC (State Commitment), where `iavl` is used for SC and `versionDB` is already implemented for SS. The index of the `fast node` is not needed in `iavl` since `versionDB` allows key-value queries for a specific version.
```

Comment:
Which versionDB? Do you mean the implementation in cronos on top of RocksDB using the (experimental) user defined timestamp feature?

Given that random access and iteration of keys from disk will perform best when data is stored in an indexed BTree I don’t know that we should commit to this rocks implementation.  Maybe it approximates a BTree, but either it seems beyond the scope of store interfaces to mandate what the SS backend will be.

Reply:
i think we should leave versiondb out of this ADR, it is orthogonal. The goal of this adr is to cleanup the api, but we are conflating two things in this adr, store v2 changes and api clean up. I think we should separate them. First api cleanup then we do the removal of fast node system

Reply:
ok, I removed the `versionDB` from this adr scope, and made small the scope of API cleanup. I assume `nodeDB` API cleaning and creating one `Tree` interface doesn't require much work. Maybe we can open another ADR for API cleanup later.

Reply:
The primary purpose of this ADR is to support store V2, but this will require much refactoring, I wanted to add some  API cleanup through this chance.